### PR TITLE
Change code deploy and update hooks to run parallel updates

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

## Summary
_[briefly summarize the changes here]_

## Need Review By (Date)
_['10/30', 'asap', etc.]_

## Urgency
_['low', 'medium', 'high', etc.]_

## Steps to Test
1. _[First testing step]_
2. ...

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)

### CODE REVIEW

File: blt/src/Blt/Plugin/Commands/HsCommands.php [hunk 1]
```diff
@@ -8,13 +8,73 @@
...
+    $site_chunks = array_chunk($sites, ceil(count($sites) / $parallel_executions));
+    $commands = [];
+    foreach ($site_chunks as $sites) {
+      $commands[] = $this->blt()
+        ->arg('humsci:update-sites')
+        ->arg(implode(',', $sites))
+        ->getCommand();
+    }
+    file_put_contents(sys_get_temp_dir() . '/update-report.txt', '');
+    $this->taskExec(implode(" &\n", $commands) . PHP_EOL . 'wait')->run();
```
Comment (joegl) [inline blt/src/Blt/Plugin/Commands/HsCommands.php:39]: Would this need two ampersands? `&&`

Comment (pookmish) [reply blt/src/Blt/Plugin/Commands/HsCommands.php:39]: 1 `&` allows parallel execution. 2 `&&` will put them into sequence, with the later only executing if the preceeding commands succeed.

Comment (joegl) [reply blt/src/Blt/Plugin/Commands/HsCommands.php:39]: TIL. Neat 👍

File: blt/src/Blt/Plugin/Commands/HsCommands.php [hunk 2]
```diff
@@ -8,13 +8,73 @@
...
+  public function updateSites($sites = NULL) {
+    $sites = $sites ? explode(',', $sites) : $this->getConfigValue('multisites');
+    foreach ($sites as $site_name) {
+      $this->switchSiteContext($site_name);
+      $result = $this->taskDrush()
+        ->drush('updb')
+        ->drush('config:import')
+        ->option('partial')
+        ->run();
+      file_put_contents(sys_get_temp_dir() . '/update-report.txt', $site_name . ($result->wasSuccessful() ? ':1' : ':0') . PHP_EOL, FILE_APPEND);
```
Comment (joegl) [inline blt/src/Blt/Plugin/Commands/HsCommands.php:75]: This assumes the `update-report.txt` file exists, but it's created in the post deploy command and this command can be run separately.

Comment (joegl) [reply blt/src/Blt/Plugin/Commands/HsCommands.php:75]: Would it hard fail if the file doesn't exist?

Comment (joegl) [reply blt/src/Blt/Plugin/Commands/HsCommands.php:75]: Nevermind we're good:

https://www.php.net/manual/en/function.file-put-contents.php

> If filename does not exist, the file is created

**Review by joegl — APPROVED** [review]
This is great. Excited to see it in action 👍 

I'm wondering if we'd maybe want to make the parallel execution number a passable option/argument with a max limit, but that could be an "in the future" thing.

**Pull request merged**